src/main/java/com/es/phoneshop/service/ProductService.java

```diff
@@ -0,0 +1,17 @@
+package com.es.phoneshop.service;
+
+import com.es.phoneshop.model.entity.Product;
+
+import java.util.List;
+
+public interface ProductService {
+    Product getProductById(Long id);
+
+    List<Product> getProductByDescription(String description);
+
+    List<Product> getProductsNotNull();
+
+    void buyProduct(Long id);
```

Comment:
what does it mean in scope of 1st task
and why buying of the product means its deletion? Deletion of product mean that you remove it from  database(in your case from list) and this product is not present for others customers
Buying means decreasing of the stock